src/pedersen.rs

```diff
@@ -117,4 +117,50 @@ mod tests {
         let v = Pedersen::<Projective>::verify(&params, &mut transcript_v, cm, proof);
         assert!(v);
     }
+
+    use crate::transcript::keccak::{tests::keccak_test_config, KeccakTranscript};
+    #[test]
+    fn test_pedersen_vector_keccak() {
```

Comment:
Since part of the logic of this test is repeated from `test_pedersen_vector`, could we abstract all these tests (together with the old `test_pedersen_vector`) to reuse the same test logic being called from different parent tests?
Eg. all of them do the same logic which can be in a test function, the only difference is how we initialize the `Transcript` obejct.
For example what is done here https://github.com/microsoft/Nova/blob/200b47edab190c06a44cbd87a63d34a444184a41/src/nifs.rs#L381 to test the same logic but with different instantiations, but in this PR case it would be different tests `test_pedersen_vector_poseidon` & `test_pedersen_vector_keccak` which initialize the different Transcripts and then call the same function `test_pedersen_vector_with` which contains the actual test logic.

Reply:
✅ https://github.com/privacy-scaling-explorations/folding-schemes/pull/27/commits/bf4cdbc3b694b606ab207b332551f9c77fae0e81